File: client/lib/model/dto/topic.dart
import 'package:json_annotation/json_annotation.dart';

part 'topic.g.dart';

@JsonSerializable()
class GetTopicDto {
  @JsonKey(name: "id", defaultValue: 0)
  int id;

  @JsonKey(name: "created_at")
  DateTime createAt;

  @JsonKey(name: "deleted_at")
  DateTime deleteAt;

  @JsonKey(name: "name")
  String name;

  @JsonKey(name: "desc")
  String desc;

  // int creator;

  @JsonKey(name: "invite_code")
  String inviteCode;

  GetTopicDto(this.id, this.createAt, this.deleteAt, this.name, this.desc,
      this.inviteCode);

  factory GetTopicDto.fromJson(Map<String, Object?> json) =>
      _$GetTopicDtoFromJson(json);
}

File: server/model/po/v1/topic.go
package po

import "time"

type SubscribedTopic struct {
	Id         uint      `json:"id" gorm:"column:id"`
	CreateAt   time.Time `json:"created_at" gorm:"column:created_at"`
	DeleteAt   time.Time `json:"deleted_at" gorm:"column:deleted_at"`
	Name       string    `json:"name" gorm:"column:name"`
	Desc       string    `json:"desc" gorm:"column:desc"`
	Creator    uint      `json:"creator" gorm:"column:creator"`
	InviteCode string    `json:"invite_code" gorm:"column:invite_code"`
}

type SubscribedMemeber struct {
	Id   uint   `json:"id"`
	Name string `json:"name"`
	Perm uint8  `json:"perm"`
}

const (
	PermMember = iota
	PermManager
	PermAdmin
)
